### PR TITLE
Added scripts and test data for json and yaml support.

### LINKED ISSUE #496: add code to deserialize json and yaml files and extract particular key value pairs
I have a messy issue-434 branch that has some perl scripts (deserialize.pl, read_json.pl, etc) along with some test metadata file in json and yaml formats. I will create a new branch for this issue, put those files on this new branch to clean it up, and submit a PR for review.

### PR DESCRIPTION
**Pull request by jasonfleming**
resolves #496

### CODE REVIEW

File: util/input/meteorology/read_json.pl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    print "null";
+}
+
+
+close(APPLOGFILE);
+1;
+
+#
+# write a log message to stderr
+sub stderrMessage () {
```
Comment (wwlwpd) [inline util/input/meteorology/read_json.pl:76]: yeah definitely put this in a module, maybe `./PERL/ASGS/Logger.pm` (i.e., `use ASGS::Logger`) - `$SCRIPTDIR/PERL` is already in `PERL5LIB`, so new custom modules just need to go in there.

**Review by wwlwpd — CHANGES_REQUESTED** [review]
see code comments

**Comment by wwlwpd** [comment]
@jasonfleming upon further review, this PR really looks like it was not fully completed. When you get a chance, please make sure the files being added in this branch are working as intended. I can't see how they are, especially `deserialize.pl` which won't even compile under `perl` for a variety of reasons.

File: util/input/meteorology/deserialize.pl [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    }
+    # slurp the file contents into a scalar variable
+    my $file_content = do { local $/; <F> };
+    close(F);
+} else {
+   &stderrMessage("ERROR","Did not provide the name of the file to be read.");
+   &appMessage("ERROR","Did not provide the name of the file to be read.");
+   exit 1;
+}
+if ( $type eq "json" ) {
```
Comment (wwlwpd) [inline util/input/meteorology/deserialize.pl:68]: `$type` is never defined.

**Review by wwlwpd — CHANGES_REQUESTED** [review]
@jasonfleming I need you to review this PR and tell me if it's incomplete. I can't guess what it is you want to do in some cases; also:

* `deserialize.pl` won't compile (see comments)
* `read_json.pl` also looks incomplete (e.g., it has `use YAML::Tiny;`)

**Comment by jasonfleming** [comment]
@wwlwpd sorry for the confusion, I will fix these issues.

**Comment by jasonfleming** [comment]
Hey @wwlwpd sorry about this PR, I started work on it in March and then set it aside. When I came back to it, I did not pick up where I left off correctly. These scripts and test files were experimental, and I had forgotten that. We do not need to move forward with these, so I am closing this PR. Thank you for reviewing this.

**Pull request closed without merging**